File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
      int count =0;
      int candidate = nums[0];
      for(int i=0; i<nums.length; i++){
        if(candidate == nums[i]){
          candidate = nums[i];
          count++;

        }
        else{
          count--;
          if(count ==0){
            candidate = nums[i];
            count =1;
          }
        }
      }
      return candidate;

        
    }
}

File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int candidate = 0;
        int rai = 0;

        
        for (int i = 0; i < nums.size(); i++) {
            if (rai == 0) {
                candidate = nums[i];
                rai = 1;
            } else if (candidate == nums[i]) {
                rai++;
            } else {
                rai--;
            }
        }

        return candidate;
    }
};
